src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,94 @@
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String DELIMITER = ",";
+    private static final int EXPECTED_PARTS = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> lines = readLines(fromFileName);
+        long[] sums = processLines(lines);
+        String report = buildReport(sums[0], sums[1]);
+        writeReport(toFileName, report);
+    }
 
+    private List<String> readLines(String fileName) {
+
+        Path path = Path.of(fileName);
+        try (BufferedReader reader = Files.newBufferedReader(path)) {
+            return reader.lines().toList();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file " + fileName, e);
+        }
     }
+
+    /**
+     * Process CSV lines and return array {supplySum, buySum}
+     */
+    private long[] processLines(List<String> lines) {
+        long supplySum = 0L;
+        long buySum = 0L;
+
+        for (String rawLine : lines) {
+            if (rawLine == null) {
+                continue;
+            }
+
+            String line = rawLine.trim();
+            if (line.isEmpty()) {
+                continue;
+            }
+
+            String[] parts = line.split(DELIMITER);
+            if (parts.length != EXPECTED_PARTS) {
+                throw new RuntimeException(
+                        "Invalid record (wrong number of fields): \"" + rawLine + "\"");
+            }
+
+            String operation = parts[0].trim();
+            String amountStr = parts[1].trim();
+
+            int amount;
+            try {
+                amount = Integer.parseInt(amountStr);
+            } catch (NumberFormatException e) {
+                throw new RuntimeException("Invalid number in record: \"" + rawLine + "\"");
```

This is a critical checklist violation. The checklist says: "Don't ignore exceptions
Leaving an empty catch block or `e.printStackTrace` here is a bad practice. Better re-throw `RuntimeException` with the original exception and some message in the parameters:" . Here you catch NumberFormatException but rethrow a new RuntimeException without passing the caught exception as the cause. Change to include the original exception, e.g. `throw new RuntimeException("Invalid number in record: \"" + rawLine + "\"", e);`.